src/fmu/dataio/providers/_filedata.py

```diff
@@ -159,3 +123,41 @@ def _sanitize_the_filestem(filestem: str) -> str:
             .replace("å", "aa")
         )
         return re.sub(r"__+", "_", filestem)
+
+
+class FileDataProvider(Provider):
```

Comment:
This is not new even though it seems like it. It just have been converted to a regular class instead of a dataclass, and a lot of the code has moved into the `SharePathConstructor`. 
This has primarily changed to use the `share_path` from the `ObjectDataProvider.`